persist/sqldb/workflow_archive.go

```diff
@@ -174,20 +172,13 @@ func (r *workflowArchive) ListWorkflows(namespace string, name string, namePrefi
 		return nil, err
 	}
 	wfs := make(wfv1.Workflows, len(archivedWfs))
-	for i, md := range archivedWfs {
-		wfs[i] = wfv1.Workflow{
-			ObjectMeta: v1.ObjectMeta{
-				Name:              md.Name,
-				Namespace:         md.Namespace,
-				UID:               types.UID(md.UID),
-				CreationTimestamp: v1.Time{Time: md.StartedAt},
-			},
-			Status: wfv1.WorkflowStatus{
-				Phase:      md.Phase,
-				StartedAt:  v1.Time{Time: md.StartedAt},
-				FinishedAt: v1.Time{Time: md.FinishedAt},
-			},
+	for i, archivedWf := range archivedWfs {
+		wf := wfs[i]
+		err = json.Unmarshal([]byte(archivedWf.Workflow), &wf)
+		if err != nil {
+			return nil, err
```

Comment:
Seems like one bad workflow in database could cause whole request to fail. It may not even be a "bad" workflow per se. It could be an older workflow that doesnt marshal to current workflow API definition (maybe because of spec change).

I wonder if it would be better to log an error and continue to the next workflow, instead of return error immediately.

Reply:
I too agree with it. UI should move on if any one element is bad

Reply:
Updated the PR to print a warning instead.